spec/virtual_includes_spec.rb

```diff
@@ -462,6 +487,24 @@
     end
   end
 
+  context ".replace_virtual_field (private)" do
+    # would have prefer target: {:books => :bookmarks} without the extra :books. but here we are
+    it "handles deep includes(:uses => :books => :bookmarks)" do
+      expect(Author.replace_virtual_fields([:book_with_most_bookmarks, :books])).to eq([{:books => :bookmarks}, :books])
+      expect(Author.replace_virtual_fields([{:book_with_most_bookmarks => {}}, {:books => {}}])).to eq([{:books => :bookmarks}, {:books => {}}])
+    end
+
+    # I was a little confused that it uses author (association) instead of authors (table name)
+    # but active record seems ok with this
```

Comment:
This is weird in code to me - it's like a self-narrative conversation?  Can we write this in third person or something for other readers?

```suggestion
    # TODO: Need to understand why this uses author (association) instead of authors (table name)
    # and why active record seems ok with this
```

Reply:
yea. Think it was just me grumbling. I deleted